src/analysis/processing/qgsalgorithmjoinbynearest.cpp

```diff
@@ -169,12 +170,21 @@ QVariantMap QgsJoinByNearestAlgorithm::processAlgorithm( const QVariantMap &para
   const QgsAttributeList fields2Fetch = fields2Indices;
 
   QgsFields outFields = QgsProcessingUtils::combineFields( input->fields(), outFields2 );
```

Comment:
Better to just use combineFields again here, i.e.:

```
  QgsFields additionalFields;
  additionalFields.append( QgsField( QStringLiteral( "n" ), QVariant::Int ) );
  additionalFields.append( QgsField( QStringLiteral( "distance" ), QVariant::Double ) );
  additionalFields.append( QgsField( QStringLiteral( "feature_x" ), QVariant::Double ) );
  additionalFields.append( QgsField( QStringLiteral( "feature_y" ), QVariant::Double ) );
  additionalFields.append( QgsField( QStringLiteral( "nearest_x" ), QVariant::Double ) );
  additionalFields.append( QgsField( QStringLiteral( "nearest_y" ), QVariant::Double ) ); 
  outFields = QgsProcessingUtils::combineFields( outFields, additionalFields );
```

Reply:
I overlooked the fact that `combineFields` already has the conflict handling logic. Thanks!